### PR TITLE
Implemented tests

### PR DESCRIPTION
**Pull request by sunknudsen**
Based on #2 (thanks to @FritsvanCampen’s contribution)

Perhaps naive suggestions… looking forward to feedback.

- Dropped Babel development dependencies given `.ts` and `.tsx` files are transpired by TypeScript when one runs `npm run code` or `npm run build`
- Added `files` property to `package.json` to not include test files during packaging
- Renamed `index.test.tsx` to `test.tsx` (personal preference)
- Incremented semver to beta release (thanks to tests)
- Applied Prettier project styling

### CODE REVIEW

File: src/test.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import reactNodeToString from "./index"
 type Test = [input: React.ReactNode, expectedOutput: string]
 
 describe("react-node-to-string", () => {
-  it("outputs correct string", () => {
+  it("extract correct string", () => {
```
Comment (FritsvanCampen) [inline src/test.tsx]: I think it should be "extracts" to get the correct grammar sentence: "it extracts correct string"

**Comment by FritsvanCampen** [comment]
> Dropped Babel development dependencies given .ts and .tsx files are transpired by TypeScript when one runs npm run code or npm run build

Build is part of publishing and can be separate from development. Not having babel makes development a little bit more difficult because you have to build in between test runs. It breaks `jest --watch` for instance. But, considering this library shouldn't ever have a large code base it doesn't really matter.

> Incremented semver to stable release

Wonderful!

File: package.json [hunk 2]
```diff
@@ -15,7 +25,11 @@
     "react"
   ],
   "devDependencies": {
+    "@babel/core": "^7.15.0",
```
Comment (FritsvanCampen) [inline package.json]: `@babel/core` isn't needed anymore.

Comment (sunknudsen) [reply package.json]: Missed that one, thanks!

File: src/test.tsx (hunk 1 above)
Comment (sunknudsen) [reply src/test.tsx]: I agree!

**Comment by sunknudsen** [comment]
> Build is part of publishing and can be separate from development.

I agree package and tests should be isolated. Full disclosure, I know little about React development.

Given `src/index.test.tsx` file was in `src` folder, file was transpiled and included in `build` folder when `npm run code` or `npm run build` ran… hence why I suggested dropping Babel dependencies given lack of isolation.

That said, I would love to learn how to better isolate tests.

Perhaps you have suggestions?

**Comment by FritsvanCampen** [comment]
How most ts repos seem to do it is to have a publish action which builds the js files and then publishes that. This means the build files are never part of the repository. I don't think it's necessary to set something like that up so manually updating the build files is fine, but that means we need to make sure our configs are set up to deal with it.

I do think it is a good idea to keep full typescript support in development - so don't test the build artifacts.
To prevent jest from running the test files in `build/` we need some config to ignore that.
Maybe something like `testMatch: [ "src/**" ]` works?

If ts dev support isn't added I would change the `test` script to: `test: "npm run build; jest build/test.js"` to make it clear how it's supposed to work.

It's up to you really, what you have now works but I think that's not how people would expect it to work.

**Comment by sunknudsen** [comment]
Thanks so much for sharing your experience.

I will try to learn more about TypeScript package development and refactor codebase when I have time.

> If ts dev support isn't added I would change the test script to: test: "npm run build; jest build/test.js" to make it clear how it's supposed to work.

👍

Do you feel comfortable using package once this PR is merged?

If so, I would merge PR and publish react-node-to-string@0.1.0.

**Comment by FritsvanCampen** [comment]
> Do you feel comfortable using package once this PR is merged?

Absolutely. Thanks!

**Comment by sunknudsen** [comment]
@FritsvanCampen Published!

**Pull request merged**